File: solutions/cpp/rotational-cipher/1/rotational_cipher.cpp
#include "rotational_cipher.h"

#include<algorithm>
#include<cctype>

using namespace std;

namespace rotational_cipher {

    

    string rotate(string plaintext, int rot){
          string result = plaintext;
          transform(result.begin(), result.end(), result.begin(), [rot](int ch){
              if (!isalpha(ch))
                  return ch;
              auto up = isupper(ch);
              ch += (rot % 26);
              if( (up && ch>'Z') || (!up && ch>'z') )
                  ch -=26;
              return ch;  
          });
        return result;
    }
}  // namespace rotational_cipher

File: solutions/python/rotational-cipher/1/rotational_cipher.py
def rotate(text, key):
    result =""
    for ch in text:
        if not ch.isalpha():
            result += ch
        else:
            new_ord = ord(ch) + (key%26)
            if (ch.isupper() and new_ord > ord('Z')) or (ch.islower() and new_ord > ord('z')):
                new_ord -= 26
            result += chr(new_ord)
    return result
            
